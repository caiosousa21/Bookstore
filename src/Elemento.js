import React, { Component } from 'react';
import Categoria from './Categoria'

class Elemento extends Component {
    buscarLista = () => {
        let url = 'https://api.nytimes.com/svc/books/v3/lists/names.json?api-key=7a7846ce976e4df6ae218249bfb3a631&list=';
        function Get(urlLista) {
            var Httpreq = new XMLHttpRequest();
            Httpreq.open('GET', urlLista, false);
            Httpreq.send(null);
            return Httpreq.responseText;
        }
        let ListaListas = JSON.parse(Get(url));
        
        let Listas =[];
        for (let i = 0; i < 4; i++) {
            Listas[i] = {a:ListaListas.results[i].list_name, b:ListaListas.results[i].list_name_encoded};
        }
        return(Listas)
    }
    
    criarCategoria = () => {
        let lista = this.buscarLista();
        let categ = [];
        for (let i = 0; i < 4; i++) {
            categ.push(<Categoria key={'categoria' + i} categoria={lista[i].a} nome={lista[i].b} clickAdicionar={this.props.clickAdicionar} />);
        }
        return categ
    }

    render() {
        return (
            <div className='Elemento'>
                {this.criarCategoria()}
            </div>
        )
    }
}

export default Elemento;