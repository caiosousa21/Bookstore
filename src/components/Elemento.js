import React, { Component } from 'react';
import Categoria from './Categoria'

class Elemento extends Component {
    criarCategoria = () => {
        let listaListas = this.props.listaListas;
        let listaLivros = this.props.listaLivros;
        let categ = [];
        if (listaListas.length < 4) {
            for (let i = 0; i < listaListas.length; i++) {
                categ.push(
                    <Categoria
                        key={'categoria' + i}
                        categoria={listaListas[i]}
                        listaLivros={listaLivros[i]}
                        clickAdicionar={this.props.clickAdicionar}
                        clickAdicionarLivro={this.props.clickAdicionarLivro}
                        indexCategoria={i}
                    />);
            }
        } else {
            for (let i = 0; i < 4; i++) {
                categ.push(
                    <Categoria
                        key={'categoria' + i}
                        categoria={listaListas[i]}
                        listaLivros={listaLivros[i]}
                        clickAdicionar={this.props.clickAdicionar}
                        clickAdicionarLivro={this.props.clickAdicionarLivro}
                        indexCategoria={i}
                    />);
            }
        }
        return categ
    }

    render() {
        return (
            <div className='Elemento'>
                {this.criarCategoria()}
            </div>
        )
    }
}

export default Elemento;