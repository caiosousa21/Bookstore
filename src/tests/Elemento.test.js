import React from 'react';
import Elemento from '../components/Elemento';
import Enzyme, { shallow, mount } from "enzyme";
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

Enzyme.configure({ adapter: new Adapter() })

describe('Elemento', () => {
    it('Renderizando Elemento', () => {
        const wrapper = shallow(<Elemento listaListas='' listaLivros='' />);
        expect(wrapper.exists()).toBe(true);
    })
    it('Testando interface com snapshots',()=>{
        const tree = renderer.create(<Elemento listaListas={[['caio']]}  listaLivros={[['caio']]}/>).toJSON();
        expect(tree).toMatchSnapshot();
    })
})