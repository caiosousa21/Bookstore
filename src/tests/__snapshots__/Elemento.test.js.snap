// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Elemento Testando interface com snapshots 1`] = `
<div
  className="Elemento"
>
  <div>
    <div
      className="Titulo"
    >
      <h2
        className="HeaderTitulo"
      >
        caio
      </h2>
      <div
        className="InserirDiv"
      >
        <button
          className="Inserir"
          onClick={[Function]}
        >
          Inserir Livro
        </button>
        <form
          className="FormInserir"
          onSubmit={[Function]}
        >
          <div>
            <input
              className="InserirTitulo"
              onChange={[Function]}
              type="text"
              value=""
            />
            <input
              className="InserirAutor"
              onChange={[Function]}
              type="text"
              value=""
            />
            <input
              className="InserirEditora"
              onChange={[Function]}
              type="text"
              value=""
            />
            <input
              className="InserirData"
              onChange={[Function]}
              type="Date"
              value=""
            />
          </div>
          <input
            type="submit"
          />
        </form>
      </div>
      <button
        className="Buscar"
      >
        Buscar Livro
      </button>
    </div>
    <hr />
    <div
      className="Categoria"
    >
      <div
        className="Livro"
      >
        <div
          className="Card"
        >
          <div
            className="Capa"
          >
            <p
              className="PCapa"
            >
              Capa
            </p>
          </div>
          <p
            className="TituloLivro"
          />
          <p
            className="Autor"
          />
          <p
            className="Editora"
          />
          <p
            className="Data"
          />
        </div>
        <div
          className="Botoes"
        >
          <button
            className="BtnDescr"
          >
            Descr
          </button>
          <button
            className="BtnAdd"
            onClick={[Function]}
          >
            Add
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
